base/statistics.jl

```diff
@@ -683,7 +683,7 @@ end
     else
         a = T(v[i])
         b = T(v[i+1])
-        return a + h*(b-a)
+        return  a + ifelse(a == b, zero(a),h*(b-a))
```

Comment:
It would be great if you could remove the extra space before `a` and add a space before `h`. That would make it consistent with the style of the rest of the program.